File: frontend/src/types/skill.ts
/**
 * Skill Types
 * Type definitions for candidate skills
 */

export interface Skill {
  id: string;
  user_id: string;
  skill: string;
  created_at: string;
  updated_at: string;
}

export interface SkillCreateRequest {
  skill: string;
}

export interface SkillsListResponse {
  skills: Skill[];
  total: number;
}

export interface ProfileUpdateRequest {
  name: string;
}

export interface PasswordChangeRequest {
  currentPassword: string;
  newPassword: string;
  confirmPassword: string;
}

File: backend/app/schemas/skill.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class SkillCreateRequest(BaseModel):
    skill: str


class SkillResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    user_id: str
    skill: str
    created_at: datetime
    updated_at: datetime


class SkillsListResponse(BaseModel):
    skills: list[SkillResponse]
    total: int
